internal/security/bootstrapper/command/setupacl/acltokens.go

```diff
@@ -411,3 +378,52 @@ func (c *cmd) setAgentToken(bootstrapACLToken BootStrapACLTokenInfo, agentTokenI
 
 	return nil
 }
+
+// createNewToken creates a new token based on the provided inputs
+// it returns the whole json string containing the token and thus can be written to the file later
+func (c *cmd) createNewToken(bootstrapACLTokenID string, createToken CreateRegistryToken) (string, error) {
+	if len(bootstrapACLTokenID) == 0 {
+		return emptyToken, fmt.Errorf("bootstrap token ID cannot be empty")
+	}
+
+	createTokenURL, err := c.getRegistryApiUrl(consulCreateTokenAPI)
+	if err != nil {
+		return emptyToken, err
+	}
+
+	jsonPayload, err := json.Marshal(&createToken)
+	c.loggingClient.Tracef("payload: %v", createToken)
+	if err != nil {
+		return emptyToken, fmt.Errorf("Failed to marshal CreatRegistryToken JSON string payload: %v", err)
+	}
+
+	req, err := http.NewRequest(http.MethodPut, createTokenURL, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return emptyToken, fmt.Errorf("Failed to prepare creat a new token request for http URL: %w", err)
+	}
+
+	req.Header.Add(consulTokenHeader, bootstrapACLTokenID)
+	req.Header.Add(clients.ContentType, clients.ContentTypeJSON)
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return emptyToken, fmt.Errorf("Failed to send create a new token request for http URL: %w", err)
+	}
+
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+
+	createTokenResp, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return emptyToken, fmt.Errorf("Failed to read create a new token response body: %w", err)
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		c.loggingClient.Info("successfully created a new registry token")
+		return string(createTokenResp), nil
```

Comment:
Add comment on what is returned here. I assume it is the actual token....

Reply:
there is already the comment in this function document, no need to repeat here again

Reply:
ok